### PR TITLE
Updating generic agent to use the new goal_object

### PR DESCRIPTION
**Pull request by ThibaultLSDC**
This will be necessary if we want to run our agent on VWA.
It slightly alters the behavior of the original generic_agent, we might want to look into it.

### CODE REVIEW

File: src/agentlab/agents/dynamic_prompting.py [hunk 1]
```diff
@@ -462,6 +468,30 @@ def __init__(self, goal, visible: bool = True, extra_instructions=None) -> None:
...
+        for image_i, image in enumerate(self.goal_images):
+            prompt.extend(
+                [
+                    {
+                        "type": "text",
+                        "text": f"({image_i+img_count+1}) input image {image_i+1} (local path: {repr(image['path'])})",
+                    },
+                    {
+                        "type": "image_url",
+                        "image_url": {"url": image["base64"]},
```
Comment (ThibaultLSDC) [inline src/agentlab/agents/dynamic_prompting.py:483]: add image detail

**Comment by ThibaultLSDC** [comment]
image
The prompt and xray now looks  like this, the goal images are given right after the goal.

File: src/agentlab/agents/generic_agent/reproducibility_agent.py [hunk 2]
```diff
@@ -125,7 +125,15 @@ def _make_agent_stats(action, agent_info, step_info, old_chat_messages, new_chat
...
-    return "\n".join(f"{m['role']} message:\n{m['content']}\n" for m in messages)
+
+    res = []
+    for m in messages:
+        if isinstance(m, list):
+            assert all(el["type"] == "text" for el in m)
+            res.append("\n".join(el["content"] for el in m))
+        else:
+            res.append(m["content"])
+    return "\n".join(res)
```
Comment (ThibaultLSDC) [inline src/agentlab/agents/generic_agent/reproducibility_agent.py:136]: @recursix could we get your opinion there ?

File: src/agentlab/agents/generic_agent/generic_agent.py [hunk 3]
```diff
@@ -74,10 +74,15 @@ def obs_preprocessor(self, obs: dict) -> dict:
     @cost_tracker_decorator
     def get_action(self, obs):
 
+        goal_object = obs.pop("goal_object", [])
+        if len(goal_object) == 0:
```
Comment (recursix) [inline src/agentlab/agents/generic_agent/generic_agent.py:78]: just remove [] and let the error raise itself

File: src/agentlab/agents/most_basic_agent/most_basic_agent.py [hunk 4]
```diff
@@ -150,3 +159,11 @@ def parser(response: str) -> tuple[dict, bool, str]:
 
 def experiment_config():
     return exp_args
```
Comment (recursix) [inline src/agentlab/agents/most_basic_agent/most_basic_agent.py:161]: ooobs

File: src/agentlab/agents/dynamic_prompting.py [hunk 5]
```diff
@@ -441,24 +444,58 @@ def __init__(self, visible: bool = True) -> None:
...
 ## Extra instructions:
 
 {extra_instructions}
-"""
+""",
+                }
+            ]
+        self._prompt = self._update_detail(self._prompt)
+
+    def _update_detail(self, prompt):
```
Comment (recursix) [inline src/agentlab/agents/dynamic_prompting.py:486]: no need to pass self._prompt

File: tests/agents/test_generic_prompt.py [hunk 6]
```diff
@@ -7,6 +7,13 @@
 from agentlab.agents.generic_agent.generic_agent_prompt import GenericPromptFlags, MainPrompt
 from agentlab.llm.llm_utils import count_tokens
 
+
+def merge_texts(texts):
```
Comment (recursix) [inline tests/agents/test_generic_prompt.py:11]: have a single one of this in llm utils

**Pull request closed without merging**